### DISCUSSION TITLE
the doppler velocity dealiasing

### DISCUSSION THREAD

**Discussion by chenuanc** [ARM-DOE/pyart#1309] [Q&A]
It seems that the doppler velocity dealiasing of my radar data is not very good, especially at a distance. What should I do more?

**Comment by chenuanc** [comment]
corrected_vel = pyart.correct.dealias_region_based(nc, interval_splits=3, vel_field='velocity', keep_original=False, nyquist_vel=nyq, centered = True)

**Comment by scollis** [comment]
Good evening @chenuanc .. I am sorry but in order to help we need a lot more information. Can you provide some plots? what is the nyquist? Is the nyquist set correctly? There is not enough information here.

**Reply by chenuanc** [reply]
There seems to be something wrong in the picture on the right.

**Reply by chenuanc** [reply]
I'm a beginner, so I use the default settings.
![pic1](https://user-images.githubusercontent.com/55877927/199034119-afa23f19-ba47-41dd-a1d0-2c8d93c3aa02.png)

**Reply by mgrover1** [reply]
@chenuanc Do you have sample data we can replicate this with? It will likely require additional tuning of the parameters.

**Reply by mgrover1** [reply]
I understand if you cannot share the data... unfortunately, without the data, this is tough to reproduce. I would suggest you explore changing the interval_split values as a start.

**Reply by rcjackson** [reply]
It looks like it's failing at the gap in coverage that's making it look like there is a ring in the middle. The region based dealiasing algorithm will not merge together regions that are split by more than 100 gates in either the range or azimuth. I'd try raising the skip_between_ray and skip_along_ray settings to see if you can get the algorithm to join that outer region in with the inner ring correctly.

**Reply by chenuanc** [reply]
Thank you all for your reply, I'll try to change these parameters.

**Reply by chenuanc** [reply]
After I changed the three parameters, I got better results, but near the gap, raising the skip_between_ray and skip_along_ray does not always have a positive effect. How can I accurately remove or change the value of those small areas?
![IMG_20221102_010923](https://user-images.githubusercontent.com/55877927/199294089-6cf47b8c-8c2a-4a17-a587-9c3beb5f01f7.png)

**Reply by mgrover1** [reply]
@chenuanc - here is an example that was shared on Twitter about how to deal with dual-PRF velocity cleaning issues https://github.com/Soudaaa/correct_catastrophic_errors/blob/main/correcting_catastrophic_errors.ipynb

**Reply by chenuanc** [reply]
Thank you. I'll check it.

**Comment by scollis** [comment]
Looks like NEXRAD to me! 
having a sounding or other velocity field as a first guess will help
https://arm-doe.github.io/pyart/_modules/pyart/correct/region_dealias.html

@rcjackson do we have an example. Even CMAC might help

**Reply by rcjackson** [reply]
I've put a reply in his main thread about tweaking the skip ray settings.

**Comment by scollis** [comment]
I think this is now really getting into radar science. Suggest you take this to:https://openradar.discourse.group/  We are all there but you will find more eyes on the data there… Looks like Dual PRF outliers to me… See https://www.osti.gov/pages/biblio/1371614  From: JJC ***@***.***>Date: Tuesday, November 1, 2022 at 12:10 PMTo: ARM-DOE/pyart ***@***.***>Cc: Scott ***@***.***>, Comment ***@***.***>Subject: Re: [ARM-DOE/pyart] the doppler velocity dealiasing (Discussion #1309)After I changed the three parameters, I got better results, but near the gap, raising the skip_between_ray and skip_along_ray does not always have a positive effect. How can I accurately remove or change the value of those small areas?—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>
